controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -141,6 +141,20 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		}
 	}
 
+	// If DataImage exists, add its ownerReference
+	dataImage := &metal3api.DataImage{}
+	err = r.Get(ctx, request.NamespacedName, dataImage)
+	if !(err != nil || ownerReferenceExists(host, dataImage)) {
+		if err := controllerutil.SetControllerReference(host, dataImage, r.Scheme()); err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not set bmh as controller, %w", err)
```

Comment:
This is unnecessarily dangerous, because things we have no control over (like e.g. something else sets an owner reference on the DataImage with the controller bit set) will prevent the BMH itself from ever getting reconciled.

Reply:
Makes sense, I think raising an error message in the logs and continuing makes sense in this scenario.